envoy/stats/store.h

```diff
@@ -123,6 +139,11 @@ class StoreRoot : public Store {
    * method would be asserted.
    */
   virtual void mergeHistograms(PostMergeCb merge_complete_cb) PURE;
+
+  /**
+   * Set the predicates to filter stats for sink.
+   */
+  virtual void setSinkPredicates(std::unique_ptr<SinkPredicates> sink_predicates) PURE;
```

Comment:
I think I understand why this is legal, but in these scenarios elsewhere in Envoy we specify `&&` here. I did not see any instances of interfaces where we pass a unique_ptr by value in an interface.

Reply:
Fixed.